ui/src/views/WorkerManager/WMEditWorkerPool/index.jsx

```diff
@@ -0,0 +1,203 @@
+
+    return (
+      <Dashboard
+        title={
+          isNewWorkerPool
+            ? 'Worker Manager: Create Worker Type'
+            : 'Worker Manager: Edit Worker Type'
+        }>
+        <TextField
+          label="Enter Worker Type Name..."
+          name="name"
+          error={
+            Boolean(workerPool.name) && !isWorkerTypeNameValid(workerPool.name)
+          }
+          onChange={this.handleInputChange}
+          fullWidth
+          value={workerPool.name}
+          margin="normal"
+        />
+
+        <TextField
+          label="Enter Worker Type Description..."
+          name="description"
+          onChange={this.handleInputChange}
+          fullWidth
+          value={workerPool.description}
+          margin="normal"
+        />
+
+        <TextField
+          label="Enter Owner's Email..."
+          name="owner"
+          error={Boolean(workerPool.owner) && !workerPool.owner.includes('@')}
+          onChange={this.handleInputChange}
+          fullWidth
+          value={workerPool.owner}
+          margin="normal"
+        />
+
+        <FormGroup>
+          <FormControlLabel
+            control={
+              <Switch
+                checked={workerPool.wantsEmail}
+                onChange={this.handleSwitchChange}
+                value="wantsEmail"
+              />
+            }
+            label="Receive emails about errors"
+          />
+        </FormGroup>
+
+        <FormGroup classes={classes.group}>
+          <FormLabel component="provider">Provider:</FormLabel>
+          <TextField
+            id="select-provider-type"
+            select
+            label="Type:"
+            helperText="Which service do you want to run your tasks in?"
+            value={workerPool.providerType}
+            name="providerType"
+            onChange={this.handleInputChange}
+            margin="normal">
+            <MenuItem value="gcp">GCP</MenuItem>
+          </TextField>
+
+          <TextField
+            label="Name:"
+            value={workerPool.providerId}
+            name="providerId"
+            onChange={this.handleInputChange}
+            margin="normal"
+          />
+
+          <List>
+            <ListItem>
+              <FormLabel component="config">Configuration:</FormLabel>
+            </ListItem>
+            <ListItem>
+              <CodeEditor
+                value={JSON.stringify(workerPool.config, null, 2)}
+                onChange={this.handleEditorChange}
+                lint
+              />
+            </ListItem>
+          </List>
+        </FormGroup>
+        <Button
+          spanProps={{ className: classes.createIconSpan }}
+          disabled={invalidProviderConfig}
+          tooltipProps={{ title: 'Save' }}
+          onClick={this.handleCreateWorkerPool}
+          classes={{ root: classes.successIcon }}
+          variant="round">
+          <CheckIcon />
+        </Button>
```

Comment:
I did this exactly for consistency: we use the same icon with a plus sign for two different actions: opening up an editor and saving the work. I propose to use two different icons for two different actions. (I volunteer to change all the icons everywhere for consistency)

Reply:
> we use the same icon with a plus sign for two different actions: opening up an editor and saving the work

That would be considered a bug. Where do we do this? Most forms show the ContentSaveIcon icon when saving from what I can remember.

Reply:
> Where do we do this?

In AWS provisioner - on `.../aws-provisioner` there's the button with a plus icon, you click on it and end up on `...aws-provisioner/create` where there's again a button with a plus icon

Reply:
I haven't checked out other forms, and now that I look at Hooks - indeed, there's a lovely discette icon - imma use that!

Reply:
> In AWS provisioner - on .../aws-provisioner there's the button with a plus icon, you click on it and end up on ...aws-provisioner/create where there's again a button with a plus icon

Right. I think the reason we don't show a save icon is because those views do not allow updating a WT if I remember correctly.

Reply:
> I think the reason we don't show a save icon is because those views do not allow updating a WT if I remember correctly.

I'd expect it to be grayed out then, rather than having a different icon